abipy/abio/factories.py

```diff
@@ -154,6 +154,34 @@ def _find_scf_nband(structure, pseudos, electrons, spinat=None):
     return int(nband)
 
 
+def _find_nscf_nband_from_gsinput(gs_input: AbinitInput) -> int:
+    """
+    Find the value of nband for a NSCF calculation based on the input of a previous gs calculation.
+
+    Args:
+        gs_input (AbinitInput): the ground state input.
+
+    Returns:
+        the value of nband.
+    """
+    scf_nband = gs_input.get("nband")
+    if scf_nband is None:
+        occopt = gs_input.get("occopt", 1)
+        tsmear = gs_input.get("tsmear", 0.01)
+        smearing = aobj.Smearing(occopt, tsmear)
+
+        # only nsppol is used in _find_scf_nband, so just set that with a meaningful value
+        nsppol = gs_input.get("nsppol", 1)
+        spin_mode = aobj.SpinMode(mode="test", nsppol=nsppol, nspinor=1, nspden=nsppol)
+
+        charge = gs_input.get("charge", 0)
+        electrons = aobj.Electrons(spin_mode=spin_mode, smearing=smearing, charge=charge)
+        spinat = gs_input.get('spinat', None)
+        scf_nband = _find_scf_nband(gs_input.structure, gs_input.pseudos, electrons, spinat)
+
+    return scf_nband + 10
```

Comment:
Having an even number of bands in the GS/NSCF part facilitates the interoperability with autoparal and paral_kgb 1 as npband must divide nband. Also

>>> When performing Gamma-only calculations ([istwfk](https://docs.abinit.org/variables/dev#istwfk) = 2), it is recommended to set bandpp = 2 (or a multiple of 2) as the time spent in FFTs is divided by two

Perhaps we can enforce an even number of bands in _find_nscf_band_from_gsinput.
This changes the previous behaviour but I don't think it's a big deal.